plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryCliProvider.kt

```diff
@@ -0,0 +1,127 @@
+
+            // otherwise we need to unpack into a file
+            project.logger.info("[sentry] Trying to load cli from $resourcePath in a temp file...")
+
+            loadCliFromResourcesToTemp(resourcePath)?.let {
+                project.logger.info("[sentry] cli Found: $it")
+                return@getSentryCliPath it
+            } ?: project.logger.warn("[sentry] Failed to load sentry-cli from resource folder")
+        }
+
+        project.logger.error("[sentry] Falling back to invoking `sentry-cli` from shell")
+        return "sentry-cli"
+    }
+
+    internal fun getSentryPropertiesPath(project: Project): String? =
+        listOf(
+            project.file("sentry.properties"),
+            project.rootProject.file("sentry.properties")
+        ).firstOrNull(File::exists)?.path
+
+    internal fun searchCliInPropertiesFile(project: Project): String? {
+        return getSentryPropertiesPath(project)?.let { propertiesFile ->
+            runCatching {
+                Properties()
+                    .apply { load(FileInputStream(propertiesFile)) }
+                    .getProperty("cli.executable")
+            }.getOrNull()
+        }
+    }
+
+    internal fun searchCliInNodeModules(rootPath: String): String? =
+        listOf(
+            "$rootPath/../node_modules/@sentry/cli/bin/sentry-cli",
+            "$rootPath/../node_modules/@sentry/cli/bin/sentry-cli.exe",
+            "$rootPath/../node_modules/sentry-cli-binary/bin/sentry-cli",
+            "$rootPath/../node_modules/sentry-cli-binary/bin/sentry-cli.exe"
+        ).firstOrNull {
+            File(it).exists()
+        }
+
+    internal fun searchCliInResources(resourcePath: String): String? {
+        val resourceURL = javaClass.getResource(resourcePath)
+        val resourceFile = resourceURL?.let { File(it.file) }
+        return if (resourceFile?.exists() == true) {
+            resourceFile.absolutePath
+        } else {
+            null
+        }
+    }
+
+    internal fun loadCliFromResourcesToTemp(resourcePath: String): String? {
+        val resourceStream = javaClass.getResourceAsStream(resourcePath)
+        val tempFile = File.createTempFile(".sentry-cli", ".exe").apply {
+            deleteOnExit()
+            setExecutable(true)
+        }
+
+        return if (resourceStream != null) {
+            FileOutputStream(tempFile).use { output ->
+                resourceStream.use { input ->
+                    input.copyTo(output)
+                }
+            }
+            tempFile.absolutePath
+        } else {
+            null
+        }
+    }
+
+    internal fun getCliSuffix(): String {
+        val osName = System.getProperty("os.name").toLowerCase()
+        val osArch = System.getProperty("os.arch")
+        return when {
+            "mac" in osName -> "Darwin-x86_64"
+            "linux" in osName -> if (osArch == "amd64") "Linux-x86_64" else "Linux-$osArch"
+            "win" in osName -> "Windows-i686.exe"
+            else -> ""
```

Comment:
should we return `null` instead? what's the advantage of returning empty if we check `isNotBlank`